ReactiveCocoa/NSObject+Intercepting.swift

```diff
@@ -0,0 +1,37 @@
+import Foundation
+import ReactiveSwift
+import enum Result.NoError
+import ReactiveCocoaUtils.Private
+
+extension Reactive where Base: NSObject {
+	/// Create a signal which sends a `next` event at the end of every invocation
+	/// of `selector` on the object.
+	///
+	/// - parameters:
+	///   - selector: The selector to observe.
+	///
+	/// - returns:
+	///   A trigger signal.
+	public func trigger(for selector: Selector) -> Signal<(), NoError> {
+		objc_sync_enter(self)
+		defer { objc_sync_exit(self) }
+
+		let map = associatedValue { _ in NSMutableDictionary() }
+
+		let selectorName = String(describing: selector) as NSString
+		if let signal = map.object(forKey: selectorName) as! Signal<(), NoError>? {
```

Comment:
It is for certain an optional of `Signal<(), NoError>`, and we do not expect it to be something else. Using `as?` would mean allowing types other than it to slip through.

https://github.com/ReactiveCocoa/ReactiveCocoa/pull/3210#discussion_r83721833